infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientRouter.java

```diff
@@ -144,11 +144,15 @@ public IClientRouter addClient(IClient client) {
         client.setRouter(this);
 
         // Iterate through all types of CorfuMsgType, registering the handler
-        client.getHandledTypes().stream()
-                .forEach(x -> {
-                    handlerMap.put(x, client);
-                    log.info("Registered client to handle messages of type {}", x);
-                });
+        try {
+            client.getHandledTypes().stream()
+                    .forEach(x -> {
+                        handlerMap.put(x, client);
+                        log.info("Registered client to handle messages of type {}", x);
+                    });
+        } catch (Exception ex) {
+            // No Legacy getHandledTypes - Ignore
+        }
```

Comment:
Is the exception expected to be thrown when getHandledTypes() returns empty set? I thought it might be better if we explicitly check `if (client.getHandledTypes().isEmpty())` and log the warn/info. I'm not 100% sure that will it throw an exception or just pass silently if the streaming collection is empty

Reply:
Not an empty set, but when it is simply called with no implementation.  If the clients don't provide a legacy message handler explicitly, then the underlying `getMsgHandler` call throws an `UnsupportedOperation` exception. A set of size zero would cause no exception.